### DISCUSSION TITLE
ESXI Compatibility

### DISCUSSION THREAD

**Discussion by muppet3000** [galexrt/dellhw_exporter#63] [Q&A]
Has anyone tried to get this library working on an ESXI server? 
I'd love to get the metrics from OSMA into Prometheus, currently I have to run a Windows VM on my esxi host just to run the Dell server so I can then point a web browser at it.
I have the ESXI 'vib' installed and running on the ESXI server, so I'm guessing that's the 'service' aspect of what's needed, but no idea if I can run the binary directly on the host and just punch a hole through the firewall.
Would be awesome if anyone has any hints or suggestions.

**Comment by galexrt** [comment]
@muppet3000 As far as I know, OMSA (omreport) requires direct access to the hosts hardware in "some way" (should be through a kernel module if I remember correctly).

So I don't think the dellhw_exporter would work for that case, unless there is some way to have the Dell OMSA components run on the ESXI host and then also putting the dellhw_exporter on it directly somehow.
